WcaOnRails/app/views/competitions/edit.html.erb

```diff
@@ -54,7 +54,11 @@
 
     <% if @competition_admin_view || !is_actually_confirmed %>
       <%= f.input :id %>
-      <%= f.input :name %>
+      <% if @competition.warnings[:name].nil? %>
+        <%= f.input :name %>
+      <% else %>
+        <%= f.input :name, wrapper_html: { class: 'has-warning' } %>
```

Comment:
Nice!! Glad you figured something out here =)

Since nil is falsey, I don't think there's actually any value in if it's nil here.

I'd probably get fancy and use a ternary here. Maybe something like the following? Your call what you think is best.

```erb
<%= f.input :name, wrapper_html: { class: @competition.warnings[:name] ? "has-warning" : "" } %>
```